WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -81,6 +81,7 @@ public class CoinJoinClient
 	private bool RedCoinIsolation { get; }
 	private int SemiPrivateThreshold { get; }
 	private TimeSpan FeeRateMedianTimeFrame { get; }
+	internal virtual Random Random { get; set; } = Random.Shared; // Mockable random used by the decomposer.
```

Comment:
I wouldn't add another `Random` to the class, instead I would inject the optional Random in the constructor and use the SecureRandom in case it is not provided.

Reply:
In that case, I need to refactor the `AmountDecomposer`. It is using Random, but not WasabiRandom. Still worth doing it?

Reply:
Yes, i think so.